### PR TITLE
Add version to docs sidebar overview item href

### PR DESCRIPTION
**Pull request by ashleemboyer**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1201903760348480/1202364251355937/f) 🎟️

## 🗒️ What

- Added some comments to separate imports in `Sidebar` in my first approach, thought they were nice so I left them in
- Updated `SidebarSidecarLayout`'s server code to determine if there is a version being viewed, and build `overviewItemHref` based on that

## 🤷 Why

- All docs `Sidebar` nav data includes versions in it, but the `overviewItemHref` had not been coded to handle that yet
- This approach taken will help in the next PR for updating the breadcrumb to handle looking at a past version

## 🛠️ How

- First, the `context.params.page` array is checked to see if there is a version in it matching any version in the `versions` `VersionSelectItem` array passed to `SidebarSidecarLayout`
- Then, if a version is found, `overviewItemHref` has the found version appended to its `href`

### Learn pages

- [ ] Vault tutorials landing: [/vault/tutorials](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/vault/tutorials)
- [ ] A Vault collection: [/vault/tutorials/getting-started](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/vault/tutorials/getting-started)
- [ ] Waypoint tutorials landing: [/waypoint/tutorials](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/waypoint/tutorials)
- [ ] A Waypoint collection: [/waypoint/tutorials/get-started-docker](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/waypoint/tutorials/get-started-docker)

### Docs pages

For the steps below for the following pages:

- [/vault/docs](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/vault/docs)
- [/vault/api-docs](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/vault/api-docs)
- [/waypoint/docs](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/waypoint/docs)
- [/waypoint/commands](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/waypoint/commands)
- [/waypoint/plugins](https://dev-portal-git-ambadd-version-to-overview-item-href-hashicorp.vercel.app/waypoint/plugins)

Steps:

- [ ] Make sure the Overview sidebar item has the correct link
- [ ] Change the version of docs being viewed
- [ ] Make sure the Overview sidebar item has the correct link

## 💭 Anything else?

Upcoming PRs that will build on this work:

- #510 
- #511

### CODE REVIEW

**Review by BRKalow — APPROVED** [review]

File: src/layouts/sidebar-sidecar/server.ts [hunk 1]
```diff
@@ -200,6 +201,26 @@ export function getStaticGenerationFunctions<
...
+      // let indexOfVersionPathPart
+      let versionPathPart
+      if (versions) {
+        pathParts.find((pathPart, index) => {
+          const matchingVersion = versions.find(
+            (version) => pathPart === version.version
+          )
+          if (matchingVersion) {
+            versionPathPart = pathPart
+            // indexOfVersionPathPart = index
```
Comment (BRKalow) [inline src/layouts/sidebar-sidecar/server.ts:218]: ⛏️  Can we remove this comment? Or is that done in one of the follow-ups?

File: src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css [hunk 2]
```diff
@@ -82,6 +82,7 @@ for further details.
 }
 
 .sidebarContentWrapper {
+  height: 100%; /* for positioning the version switcher */
```
Comment (BRKalow) [inline src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css:85]: ❓  was this intended to be included in this PR?

File: src/layouts/sidebar-sidecar/server.ts (hunk 1 above)
Comment (ashleemboyer) [reply src/layouts/sidebar-sidecar/server.ts:218]: It's done in one of the follow-ups, I noted this in the TODO above this section 😊

File: src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css (hunk 2 above)
Comment (ashleemboyer) [reply src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css:85]: Yes!

**Review by zchsh — APPROVED** [review]

File: src/layouts/sidebar-sidecar/server.ts (hunk 1 above)
Comment (BRKalow) [reply src/layouts/sidebar-sidecar/server.ts:218]: 👀  Sure enough!

**Pull request merged**